dali/python/nvidia/dali/types.py

```diff
@@ -147,6 +147,17 @@ class ScalarConstant(object):
     """
     def __init__(self, value, dtype=None):
         self.shape = []
+        value_dtype = getattr(value, "dtype", None)  # handle 0D tensors and numpy scalars
+        if value_dtype is not None:
+            dali_type = to_dali_type(value.dtype)
+            if dali_type in _int_types:
+                value = int(value)
```

Comment:
is this int32 or int64? are we OK with changing types?

Reply:
Python's `int` is 64-bit, so it should cover everything except UINT64 values above 2^63 - I think we can live with it.